src/storage/remote/protocol.c

```diff
@@ -138,197 +139,351 @@ storageRemoteProtocolInfoListCallback(void *server, const StorageInfo *info)
 }
 
 /**********************************************************************************************************************************/
-bool
-storageRemoteProtocol(const String *command, const VariantList *paramList, ProtocolServer *server)
+void
+storageRemoteFeatureProtocol(const VariantList *paramList, ProtocolServer *server)
 {
     FUNCTION_LOG_BEGIN(logLevelDebug);
-        FUNCTION_LOG_PARAM(STRING, command);
         FUNCTION_LOG_PARAM(VARIANT_LIST, paramList);
         FUNCTION_LOG_PARAM(PROTOCOL_SERVER, server);
     FUNCTION_LOG_END();
 
-    ASSERT(command != NULL);
+    ASSERT(paramList == NULL);
+    ASSERT(server != NULL);
 
-    // Determine which storage should be used
-    const Storage *storage = protocolStorageTypeEnum(
-        cfgOptionStr(cfgOptRemoteType)) == protocolStorageTypeRepo ? storageRepoWrite() : storagePgWrite();
-    StorageInterface interface = storageInterface(storage);
-    void *driver = storageDriver(storage);
+    MEM_CONTEXT_TEMP_BEGIN()
+    {
+        // Get storage based on remote type
+        const Storage *storage =
+            protocolStorageTypeEnum(cfgOptionStr(cfgOptRemoteType)) == protocolStorageTypeRepo ?
+                storageRepoWrite() : storagePgWrite();
```

Comment:
The remote always has --repo and --pg set so it can just use the default. Each remote addresses one and only one pg/repo storage. The logic for this is in protocol/helper.c.